### PR TITLE
Engine: Misc helpers

### PR DESCRIPTION
**Pull request by Popov72**
Miscellaneous changes I needed for my pet project.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14546/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14546/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14546/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14546/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14546/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14546/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14546/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14546/merge#BCU1XR#0

**Review by RaananW — APPROVED** [review]

**Review by sebavan — COMMENTED** [review]
Wondering if _ratio should be renamed ?

File: packages/dev/core/src/Rendering/geometryBufferRenderer.ts [hunk 1]
```diff
@@ -343,7 +343,7 @@ export class GeometryBufferRenderer {
      * How big is the buffer related to the main canvas.
      */
     public get ratio(): number {
-        return this._ratio;
+        return typeof this._ratio === "object" ? 1 : this._ratio;
```
[the remark this answers is not in the document]
Comment (Popov72) [reply packages/dev/core/src/Rendering/geometryBufferRenderer.ts:346]: Never :)

We never call it ourselves, and I can't hardly think why someone would call it...

**Pull request merged**